diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/DataClassesRule.kt

```diff
@@ -64,9 +68,47 @@ class DataClassesRule(private val configRule: List<RulesConfig>) : Rule("data-cl
                     && classBody?.getAllChildrenWithType(FUN)?.isEmpty() ?: false
                     && getFirstChildWithType(SUPER_TYPE_LIST) == null
         }
-        return classBody?.getAllChildrenWithType(FUN)?.isEmpty() ?: false && getFirstChildWithType(SUPER_TYPE_LIST) == null
+        return classBody?.getAllChildrenWithType(FUN)?.isEmpty() ?: false
+                && getFirstChildWithType(SUPER_TYPE_LIST) == null
+                // if there is any prop with logic in accessor then don't recommend to convert class to data class
+                && if (classBody != null) areGoodProps(classBody) else true
     }
 
+    /**
+     * Checks if any property with accessor contains logic in accessor
+     */
+    private fun areGoodProps(node: ASTNode): Boolean {
+        val propertiesWithAccessors = node.getAllChildrenWithType(PROPERTY).filter { it.hasChildOfType(PROPERTY_ACCESSOR) }
+
+        if (propertiesWithAccessors.isNotEmpty()) {
+            propertiesWithAccessors.forEach {
+                val accessors = it.getAllChildrenWithType(PROPERTY_ACCESSOR)
+
+                if (!areGoodAccessors(accessors))
+                    return false
```

Comment:
```suggestion
            propertiesWithAccessors.any {
                val accessors = it.getAllChildrenWithType(PROPERTY_ACCESSOR)

                !areGoodAccessors(accessors)
```
I'd rater use operation on collection and then return it's result than place an additional `return` inside the loop.